tooling/SecretManager.VisualStudio/ProjectLocalSecretsManager.cs

```diff
@@ -0,0 +1,117 @@
+        {
+            EnsureKeyNameIsValid(name);
+            var store = await GetStoreAsync();
+            if (store.ContainsKey(name))
+            {
+                throw new ArgumentException("A secret with this name already exists.", nameof(name));
+            }
+
+            store.Set(name, value);
+            cancellationToken.ThrowIfCancellationRequested();
+            store.Save();
+        }
+
+        public async Task SetSecretAsync(string name, string value, CancellationToken cancellationToken = default)
+        {
+            EnsureKeyNameIsValid(name);
+
+            var store = await GetStoreAsync();
+
+            store.Set(name, value);
+            cancellationToken.ThrowIfCancellationRequested();
+            store.Save();
+        }
+
+        public async Task<string> GetSecretAsync(string name, CancellationToken cancellationToken = default)
+        {
+            EnsureKeyNameIsValid(name);
+            var store = await GetStoreAsync();
+            return store[name];
+        }
+
+        public async Task<IReadOnlyCollection<string>> GetSecretNamesAsync(CancellationToken cancellationToken = default)
+        {
+            var store = await GetStoreAsync();
+            return store.ReadOnlyKeys;
+        }
+
+        public async Task<IReadOnlyDictionary<string, string>> GetSecretsAsync(CancellationToken cancellationToken = default)
+        {
+            return await GetStoreAsync();
+        }
+
+        public async Task<bool> RemoveSecretAsync(string name, CancellationToken cancellationToken = default)
+        {
+            EnsureKeyNameIsValid(name);
+
+            var store = await GetStoreAsync();
+
+            var result = store.Remove(name);
+
+            cancellationToken.ThrowIfCancellationRequested();
+            store.Save();
+
+            return result;
+        }
+
+        private void EnsureKeyNameIsValid(string name)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (name.Length == 0)
+            {
+                throw new ArgumentException(nameof(name));
+            }
+        }
+
+        private async Task<SecretsStore> GetStoreAsync()
+        {
+            var userSecretsId = await _propertiesProvider.GetCommonProperties().GetEvaluatedPropertyValueAsync(UserSecretsPropertyName);
+
+            if (string.IsNullOrEmpty(userSecretsId))
+            {
+                // TODO how should this be handled?
+                throw new InvalidOperationException("UserSecrets cannot be read or altered on this project because 'UserSecretsId' is not set");
+            }
+
+            // TODO figure out why this API causes compiler error:
+            // "Cannot find the interop type that matches the embedded interop type 'Microsoft.VisualStudio.Shell.Interop.IVsTask'."
+             //await TaskScheduler.Default;
```

Comment:
This is a conflict between `Microsoft.VisualStudio.Shell.Task` and `System.Threading.Tasks.Task`. I make it a point to just alias everything I use from the VS shell namespace if I'm mixing shell and async code.

Reply:
I have this at the top of the file: `using Task = System.Threading.Tasks.Task;`. Is that the issue?

Reply:
I'd try nuking the `using Microsoft.VisualStudio.Shell` and see if that fixes this problem (while causing others)